test/e2e/runner/cli_provisioner.go

```diff
@@ -87,15 +87,30 @@ func (cli *CLIProvisioner) Run() error {
 	return errors.New("Unable to run provisioner")
 }
 
-func createSaveSSH(outputPath string, privateKeyName string) (string, error) {
+func createSaveSSH(outputPath string, privateKeyName string, existingPrivateKeyPath string) (string, error) {
 	os.Mkdir(outputPath, 0755)
 	keyPath := filepath.Join(outputPath, privateKeyName)
 	cmd := exec.Command("ssh-keygen", "-f", keyPath, "-q", "-N", "", "-b", "2048", "-t", "rsa")
 
+	if existingPrivateKeyPath != "" {
+		err := os.Rename(existingPrivateKeyPath, keyPath)
+		if err != nil {
+			return "", errors.Wrapf(err, "Error while trying to move private ssh key")
+		}
+		cmd = exec.Command("ssh-keygen", "-y", "-f", keyPath)
+	}
+
 	util.PrintCommand(cmd)
-	out, err := cmd.CombinedOutput()
-	if err != nil {
-		return "", errors.Wrapf(err, "Error while trying to generate ssh key\nOutput:%s", out)
+	out, err2 := cmd.CombinedOutput()
```

Comment:
Instead of declaring a few `err#` variables, it would be more idiomatic to declare `var err error` at the beginning of the func and reuse it.

Reply:
Updated